### PR TITLE
Fix Spring integration tests for sovereign clouds

### PR DESCRIPTION
**Pull request by saragluna**
# Description

There's a bug in Spring Cloud Azure autoconfigure when copying properties from `spring.cloud.azure.eventhubs.xxx` to `spring.cloud.azure.eventhubs.producer|consuemr|processor.xxx`, if the latter already has a default value, the copy won't be effective. 

This PR is to fix this bug and refactor IT modules to reduce the surfacing ENV variables. 

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by saragluna** [comment]
Hi @benbp, I ran into an issue of failing to provision the cosmos database in Azure China Cloud. 

I tried to run `./eng/common/TestResources/New-TestResources.ps1` on my local machine or copy the ARM template to Azure Portal, and they all succeeded.  

The ARM template file is like https://github.com/Azure/azure-sdk-for-java/blob/c8cb5c1b503367423c93b837057f28308ef920f9/sdk/spring/spring-cloud-azure-integration-tests/test-resources/cosmos/test-resources.json#L1-L211

and you can find the error message here https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1827368&view=logs&j=f6cd9d1b-db6b-5aad-05d5-bfb021ebbd54&t=aa69fa7e-1a04-55e2-c533-3157be69fd83&l=6674. 

Since I don't have permission to access the logs correlated to the `CorrelationId: 739c56db-c03f-4434-96bb-652ca9ed9731`. So could you help take a look at this?

**Comment by benbp** [comment]
@saragluna the first error I'm seeing from that deployment is [here](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1827368&view=logs&j=f6cd9d1b-db6b-5aad-05d5-bfb021ebbd54&t=aa69fa7e-1a04-55e2-c533-3157be69fd83&l=2045)

```
"message": "Database account creation failed. Operation Id: 43d9739f-845d-4422-b08a-c524aaa47472, Error : Service is currently unavailable. More info: https://aka.ms/cosmosdb-tsg-service-unavailable
```

We have been having some networking issues with our agents due to a new firewall that we are working on. My first suspicion is the error is either due to that, or a transient issue with the cosmosdb service in the china cloud. Was there much of a difference between the tests you ran on Wednesday (august 31st at 6:17am UTC) vs. the most recent run? Because the Wednesday tests looked to succeed for most runs except macOS.

**Comment by saragluna** [comment]
> @saragluna the first error I'm seeing from that deployment is [here](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1827368&view=logs&j=f6cd9d1b-db6b-5aad-05d5-bfb021ebbd54&t=aa69fa7e-1a04-55e2-c533-3157be69fd83&l=2045)
> 
> ```
> "message": "Database account creation failed. Operation Id: 43d9739f-845d-4422-b08a-c524aaa47472, Error : Service is currently unavailable. More info: https://aka.ms/cosmosdb-tsg-service-unavailable
> ```
> 
> We have been having some networking issues with our agents due to a new firewall that we are working on. My first suspicion is the error is either due to that, or a transient issue with the cosmosdb service in the china cloud. Was there much of a difference between the tests you ran on Wednesday (august 31st at 6:17am UTC) vs. the most recent run? Because the Wednesday tests looked to succeed for most runs except macOS.

Thanks @benbp. Yes, there was a difference between them. The one on Wednesday I disabled the cosmos provision, and I kept seeing this error for 3 weeks.

**Comment by saragluna** [comment]
Azure Pipelines failed to run 1 pipeline(s).

**Comment by saragluna** [comment]
@yiliuTo, @Netyyyy, @moarychan please help review this PR.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/eventhubs/properties/AzureEventHubsProperties.java [hunk 1]
```diff
@@ -43,6 +44,11 @@ public class AzureEventHubsProperties extends AzureEventHubsCommonProperties
     private final Consumer consumer = new Consumer();
     private final Processor processor = new Processor();
 
+    public AzureEventHubsProperties() {
+        this.setDomainName(DEFAULT_DOMAIN_NAME);
```
[the remark this answers is not in the document]
Comment (saragluna) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/eventhubs/properties/AzureEventHubsProperties.java:48]: yeah, that is the concern. We should I think.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/eventhubs/AzureBlobCheckpointStoreConfiguration.java [hunk 2]
```diff
@@ -92,16 +90,7 @@ BlobServiceClientBuilderFactory eventHubProcessorBlobServiceClientBuilderFactory
...
-        AzureEventHubsProperties.Processor.BlobCheckpointStore result = new AzureEventHubsProperties.Processor
-            .BlobCheckpointStore();
-        AzureEventHubsProperties.Processor.BlobCheckpointStore csProperties = ehProperties.getProcessor()
-                                                                                          .getCheckpointStore();
-
-        mergeAzureCommonProperties(ehProperties, csProperties, result);
-        BeanUtils.copyProperties(csProperties, result);
-
-        return result;
+        return ehProperties.buildProcessorProperties().getCheckpointStore();
```
[the remark this answers is not in the document]
Comment (saragluna) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/eventhubs/AzureBlobCheckpointStoreConfiguration.java]: removed

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/eventhubs/properties/AzureEventHubsProperties.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (saragluna) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/eventhubs/properties/AzureEventHubsProperties.java:48]: updated

**Review by yiliuTo — APPROVED** [review]
LGTM

**Pull request merged**